micro-rdk-installer/src/main.rs

```diff
@@ -130,7 +135,7 @@ fn write_credentials_to_app_binary(
         .open(binary_path)
         .map_err(Error::FileError)?;
     let file_len = app_file.metadata().map_err(Error::FileError)?.len();
-    if (file_len - nvs_start_address) <= nvs_size {
+    if (nvs_start_address + nvs_size) >= file_len {
```

Comment:
right the total hardcoded length is 4MB right?

Reply:
the hardcoded length of the binary?

Reply:
sorry I missed read the code. I actually don't quite understand the test here, if I interpret correctly it would trigger is nvs_start_address + nvs_size > 4MB (the resulting binary max size)

Reply:
this is for the (admittedly horrific scenario) that we have a bad binary and the end of the file cuts off in the middle of what would be the NVS partition.